code/game/objects/items/RCD.dm

```diff
@@ -554,28 +573,63 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
 	 *RCD_WALLFRAME is also returned as the mode when upgrading apc, airalarm, firealarm using simple circuits upgrade
 	 */
 	if(rcd_results["mode"] != RCD_WALLFRAME && rcd_results["mode"] != RCD_DECONSTRUCT)
-		//if we are trying to build a window on top of a grill [only if there isnt already a window there] then we skip this check because thats normal behaviour
-		if(rcd_results["mode"] == RCD_WINDOWGRILLE && locate(/obj/structure/grille, target_turf) && (!locate(/obj/structure/window, target_turf) && !locate(/obj/structure/window/reinforced, target_turf)))
-			//no checks proceed
+		//if we are trying to build a window on top of a grill then we skip this check because thats normal behaviour
+		if(rcd_results["mode"] == RCD_WINDOWGRILLE && istype(A, /obj/structure/grille))
+			//if a player builds a grill on top of himself manually with rods he can't finish the window if he is still in the grill.
+			var/list/structures_to_ignore
+			if(window_type == /obj/structure/window/fulltile || window_type == /obj/structure/window/reinforced/fulltile) //if we are trying to build an fultile window we only ignore the grill but other directional windows on the grill can block its construction
+				structures_to_ignore = list(/obj/structure/grille)
+			else //for normal directional windows we ignore other directional windows as they can be in diffrent directions on the grill. There is a later check during construction to deal with those
+				structures_to_ignore = list(/obj/structure/grille, /obj/structure/window, /obj/structure/window/reinforced)
+			if(is_turf_blocked(the_turf = target_turf, ignore_mobs = FALSE, ignored_atoms = structures_to_ignore))
```

Comment:
```suggestion
				structures_to_ignore = list(/obj/structure/grille, /obj/structure/window, /obj/structure/window/reinforced)

			if(is_turf_blocked(the_turf = target_turf, ignore_mobs = FALSE, ignored_atoms = structures_to_ignore))
```

Newline for clarity because this is all running together in my head